.ci/commits-check.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash -e
+
+echo "** Running commits message check..."
+exit_code=0
+readarray -t commits <<< $(git log --no-merges -n "${COMMIT_COUNT}" --format='%H')
```

Comment:
Here I am not sure it can correctly collect all submitted commits. When the patch has more than 1 commit and if the commit date of the first commit is very old, not sure how the `actions/checkout` action does the submission process work, rebase or merge? So if we see the result is wrong, we can consider using event.pull_request.commits, like this: https://api.github.com/repos/autotest/tp-qemu/pulls/3668/commits

Reply:
Why do not we use the pull_request.commits directly as we can sure it contain the correct commit info, `author` and `message` that include `Signed-off-by` ?

Reply:
`event.pull_request.commits` returns the API url, not a dict, so if we use it, we need to use `jq` to parse the data.

Reply:
It it your concern that the current operating system Ubuntu may doesn't include the `jp` package by default ?

Reply:
No, the issue is if use `event.pull_request.commits`, the logic will be more complicated, I cannot find many advantages to using it, because I don't see a clear problem with the current solution, and API can only check the message, it binds the header and body together, that is harder to handle than git log.

Reply:
Just have a quick loop with `Check out repository code` step.
Take  https://github.com/autotest/tp-qemu/pull/3131/commits for an example, the key step in my opinion is:
```
/usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/* +674d3b51b4fd0e6532c4729bc258b560b859ed78:refs/remotes/pull/3131/merge
```

Finally, commits are order with the commit time, not on the top, i am afraid the initial issue you mentioned is true :(

Reply:
Your branch was using checkout@v2, but now I upgrade it to v3, the workflow is a bit different. With checkout@v3, the commands in #3131 should be:
```
git rev-parse --verify --quiet c21aae3dd3d27ef4558a43d47a2c1bb67183422c^{object}
git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules origin c21aae3dd3d27ef4558a43d47a2c1bb67183422c
git checkout --progress --force c21aae3dd3d27ef4558a43d47a2c1bb67183422c
```

And output of git log is:
```
# git log --no-merges -2
commit c21aae3dd3d27ef4558a43d47a2c1bb67183422c (HEAD)
Author: Yanan Fu <yfu@redhat.com>
Date:   Mon May 8 16:50:39 2023 +0800

    tmp test

    Signed-off-by: Yanan Fu <yfu@redhat.com>

commit 39d6f736c9435df98b9a4059f6ad41ce4253fcd1
Author: Yanan Fu <yfu@redhat.com>
Date:   Tue Dec 14 22:35:46 2021 +0800

    trans_hugepage_defrag:fix AttributeError,refactor it to adopt SysFS

    1. Fix AttributeError:'TransparentHugePageConfig' object has no
    attribute 'value_listed'.
    2. Refactor it to adopt SysFs() and ProcFs() API to get and set fs value.
    3. Disable this case for s390x

    Signed-off-by: Yanan Fu <yfu@redhat.com>
```

So, the pipeline works perfectly. I guess `@v2` uses REBASE but `@v3` uses MERGE.

Reply:
I can not get your point well,  this one is `v3`:
https://github.com/autotest/tp-qemu/actions/runs/4911150118/jobs/8768926525?pr=3668

But result is same with before as my understanding.
Could you please explain more about the command you got above ? Thanks!

Reply:
I can not get your point well, this is for `v3`:
https://github.com/autotest/tp-qemu/actions/runs/4911150118/jobs/8768926525?pr=3668
But the command is same with before.

Could you please explain more about how you get the command in you comment above ? Thanks!

Reply:
Also this: `ref: ${{ github.event.pull_request.head.sha }}`
https://github.com/marketplace/actions/checkout#Checkout-pull-request-HEAD-commit-instead-of-merge-commit

Reply:
I see, thanks for you info, it make things more easy,  ack